nomad/node_endpoint.go

```diff
@@ -479,6 +486,37 @@ func (n *Node) createNodeEvals(nodeID string, nodeIndex uint64) ([]string, uint6
 		evalIDs = append(evalIDs, eval.ID)
 	}
 
+	var sysJobs []*structs.Job
+	if nextJob != nil {
+		sysJobs = append(sysJobs, nextJob.(*structs.Job))
+		for job := sysJobsIter.Next(); job != nil; job = sysJobsIter.Next() {
+			sysJobs = append(sysJobs, job.(*structs.Job))
+		}
+	}
```

Comment:
Maybe just move this above the fast-path so you don't have to save the first job or duplicate the append().